datafusion/core/src/physical_planner.rs

```diff
@@ -1916,6 +1916,7 @@ pub fn create_aggregate_expr_and_maybe_filter(
     // unpack (nested) aliased logical expressions, e.g. "sum(col) as total"
     let (name, e) = match e {
         Expr::Alias(Alias { expr, name, .. }) => (name.clone(), expr.as_ref()),
+        Expr::AggregateFunction(_) => (e.display_name().unwrap_or(physical_name(e)?), e),
```

Comment:
How about we call `display_name` in `physical_name` like what ScalarFunction does. 
And even more, I think there are no such difference between logical name and physical name, maybe we can have one name for these two 🤔

Reply:
It seems CastExpr is different in `write_name` and `Expr's Display` 🤔

Reply:
> It seems CastExpr is different in `write_name` and `Expr's Display` 🤔
> 
> And, there is an issue that expect CastExpr be shown in `write name`
> 
> #10274
> 
> CastExpr is removed in #3222, but I didn't find the rationale of doing this
> 
> If the issue is due to the schema change, then it will be quite tricky #10276
> 
> We can reuse `display_name` in `physical_name` to fix the regression first

Yes, I think calling display_name like physical name is a good idea, I tried it before, but there are some tests failed, there might be some reason that the display_name is different from physical name, I think for this issue, only change AggregateFunction is enough, I can take other related issue and fix them in the following PR, Thanks for review !